core/string/ustring.cpp

```diff
@@ -4693,7 +4693,7 @@ bool String::is_valid_string() const {
 	int l = length();
 	const char32_t *src = get_data();
 	bool valid = true;
-	for (int i = 0; i < l; i++) {
+	for (int i = 0; i < l && valid; i++) {
 		valid = valid && (src[i] < 0xd800 || (src[i] > 0xdfff && src[i] <= 0x10ffff));
```

Comment:
```suggestion
		valid = src[i] < 0xd800 || (src[i] > 0xdfff && src[i] <= 0x10ffff);
```
Simplify even further